runtime/lua/vim/lsp/buf.lua

```diff
@@ -162,14 +178,18 @@ end
 
 ---@param method vim.lsp.protocol.Method.ClientToServer.Request
 ---@param opts? vim.lsp.LocationOpts
-local function get_locations(method, opts)
+---@param callback? fun(errors?: string|lsp.ResponseError[])
+local function get_locations(method, opts, callback)
   opts = opts or {}
+
   local bufnr = api.nvim_get_current_buf()
   local win = api.nvim_get_current_win()
 
   local clients = lsp.get_clients({ method = method, bufnr = bufnr })
   if not next(clients) then
-    vim.notify(lsp._unsupported_method(method), vim.log.levels.WARN)
+    local msg = lsp._unsupported_method(method)
+    vim.notify(msg, vim.log.levels.WARN)
```

Comment:
Maybe this shouldn't be called if a callback is provided?

Reply:
How would we distinguish the difference between receiving no responses and no requests being made?

Reply:
I don't understand, I'm suggesting:

```lua
if not callback then
  vim.notify(msg, vim.log.levels.WARN)
end
```

Basically if the user has provided a continuation and thus is probably calling this function in a script, we should minimize effects like `vim.notify()`.

Reply:
Oh I see. I thought you were suggesting that we should let the callback figure out how to deal with the case of now clients supporting the method, and I was saying how there might not be enough information for it to know when that's the case.